packages/flutter/lib/logging.dart

```diff
@@ -0,0 +1,160 @@
+/// cost of logging calls can be further mitigated at call sites by invoking
+/// them in a function that is only evaluated in profile or debug modes. For
+/// example,
+///
+/// ```dart
+/// profile(() {
+///   _xLogger.info('X is starting...');
+///   return true;
+/// }());
+/// ```
+///
+/// In this case, logging is entirely ignored in release mode and no
+/// performance penalty is paid.
+///
+/// Clients interact with the loggers via VM Services.  Specifically,
+///
+///   * 'loggers' : provides an enumeration of available loggers
+///   * 'logger.subscribe' : supports logger subscriptions
+///
+/// See <https://github.com/dart-lang/logging> for more details on
+/// `package:logging`.
+///
+/// See <https://api.dartlang.org/stable/2.0.0/dart-developer/log.html> for more
+/// on the `log` function.
+///
+library logging;
+
+import 'dart:developer' as dev;
+
+import 'package:flutter/src/logging/logging.dart';
+import 'package:logging/logging.dart';
+
+/// A Logger object is used to log messages for specific systems or components.
+class Logger {
+  static int _sequenceNumber = 0;
+
+  /// Logger name.
+  final String name;
+
+  /// Logger description.
+  final String description;
+
+  /// Singleton constructor. Calling `Logger(name)` returns the same
+  /// actual instance whenever it is called with the same string name.
+  factory Logger(String name, {String description}) =>
+      _getOrRegisterLogger(name, description: description ?? '');
+
+  Logger._(this.name, {this.description});
+
+  static Logger _getOrRegisterLogger(String name, {String description}) =>
+      LoggingService.instance.loggers.putIfAbsent(
+          name, () => new Logger._(name, description: description));
+
+  /// The fully qualified name of this logging stream.
+  String get fullName => name;
+
+  /// Log message at level [Level.CONFIG].
+  void config(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.CONFIG, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.FINE].
+  void fine(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.FINE, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.FINER].
+  void finer(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.FINER, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.FINEST].
+  void finest(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.FINER, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.INFO].
+  void info(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.INFO, message, error, stackTrace);
+  }
+
+  /// Log [message] at the given [level] with optional [error] and [stackTrace].
+  void log(Level level, dynamic message,
+      [Object error, StackTrace stackTrace]) {
+    assert(level != null);
+    assert(message != null);
+    // If there are no subscribers, don't report.
```

Comment:
I think the `log()` method should always log, and that the higher level (individual stream objects) should be the ones to query about whether logging has been enabled for that stream.